docs/source/plugin/anatomy.rst

```diff
@@ -0,0 +1,345 @@
+         bot.reply('Hello!')
+
+
+Plugin jobs
+===========
+
+Another feature available to plugins is the ability to define
+:term:`Plugin job`. It is a Python callable decorated with
+:func:`sopel.plugin.interval` and that executes every period of time.
+
+A job follows this interface:
+
+.. py:function:: plugin_job(bot)
+
+   :param bot: the bot's instance
+   :type bot: :class:`sopel.bot.Sopel`
+
+.. note::
+
+   Note that the name can be anything, and it doesn't have to be called
+   ``plugin_job``::
+
+      from sopel import plugin
+
+      @plugin.interval(5)
+      def spam_every_5s(bot):
+          if "#here" in bot.channels:
+              bot.say("It has been five seconds!", "#here")
+
+
+.. important::
+
+   A job may execute while the ``bot`` is **not** connected, and it must not
+   assume any network access.
+
+
+Plugin setup & shutdown
+=======================
+
+When loading and unloading plugins, a plugin can perform setup and shutdown
+actions. For that purpose, a plugin can define optional functions named
+``setup`` and ``shutdown``. There can be one and only one function with each
+name for a plugin.
+
+Setup
+-----
+
+The ``setup`` function must follow this interface:
+
+.. py:function:: setup(bot)
+
+   :param bot: the bot's instance
+   :type bot: :class:`sopel.bot.Sopel`
+
+This function is optional. If it exists, it will be called while the plugin is
+being loaded. The purpose of this function is to perform whatever actions are
+needed to allow a plugin to function properly (e.g, ensuring that the
+appropriate configuration variables exist and are set). Note that this normally
+occurs prior to connection to the server, so the behavior of the messaging
+functions on the :class:`sopel.bot.Sopel` object it's passed is undefined and
+is likely to fail.
+
+Throwing an exception from this function will stop Sopel from loading the
+plugin, and none of its :term:`rules <Rule>` or :term:`jobs <Plugin job>` will
+be registered. The exception will be caught, an error message logged, and Sopel
+will try to load the next plugin.
+
+This is useful when requiring the presence of configuration values (by raising
+a :exc:`~sopel.config.ConfigurationError` error) or making other environmental
+requirements (dependencies, folder, access right, and so on).
+
+The bot will not continue loading plugins or connecting during the execution of
+this function. As such, an infinite loop (such as an unthreaded polling loop)
+will cause the bot to hang.
+
+Shutdown
+--------
+
+The ``shutdown`` function must follow this interface:
+
+.. py:function:: shutdown(bot)
+
+   :param bot: the bot's instance
+   :type bot: :class:`sopel.bot.Sopel`
+
+This function is optional. If it exists, it will be called while the bot
+is shuting down. Note that this normally occurs after closing connection
+to the server, so the behavior of the messaging functions on the
+:class:`sopel.bot.Sopel` object it's passed is undefined and is likely to fail.
```

Comment:
```suggestion
:class:`sopel.bot.Sopel` object it's passed is undefined and they are likely to fail.
```

I sense this might cause an orphan line, but oh well.